In [10]:
import pandas as pd
wv = pd.read_excel("../../data/Work Values.xlsx")

print(wv.shape)

print(wv.head())

print(wv["Scale ID"].value_counts())

(7866, 9)
  O*NET-SOC Code             Title Element ID        Element Name Scale ID  \
0     11-1011.00  Chief Executives    1.B.2.a         Achievement       EX   
1     11-1011.00  Chief Executives    1.B.2.b  Working Conditions       EX   
2     11-1011.00  Chief Executives    1.B.2.c         Recognition       EX   
3     11-1011.00  Chief Executives    1.B.2.d       Relationships       EX   
4     11-1011.00  Chief Executives    1.B.2.e             Support       EX   

  Scale Name  Data Value     Date Domain Source  
0     Extent        6.33  06/2008       Analyst  
1     Extent        6.33  06/2008       Analyst  
2     Extent        7.00  06/2008       Analyst  
3     Extent        5.00  06/2008       Analyst  
4     Extent        5.33  06/2008       Analyst  
Scale ID
EX    5244
VH    2622
Name: count, dtype: int64


In [11]:
wv = wv[wv["Scale ID"] == "EX"].copy()
print(wv.shape)  # should be (5244, 9)

(5244, 9)


In [12]:
wv_wide = wv.pivot(index=["O*NET-SOC Code", "Title"], columns="Element Name", values="Data Value").reset_index()
wv_wide.columns.name = None

In [13]:
print(wv_wide.shape)  
wv_wide.head()

(874, 8)


,O*NET-SOC Code,Title,Achievement,Independence,Recognition,Relationships,Support,Working Conditions
0,11-1011.00,Chief Executives,6.33,7.00,7.00,5.00,5.33,6.33
1,11-1011.03,Chief Sustainability Officers,6.67,6.67,6.00,5.00,3.33,6.33
2,11-1021.00,General and Operations Managers,5.33,6.00,5.67,6.33,4.67,6.00
3,11-1031.00,Legislators,5.33,5.00,5.00,5.67,4.00,4.33
4,11-2011.00,Advertising and Promotions Managers,5.33,5.33,5.33,5.00,4.00,5.33


In [14]:
print(wv_wide.isnull().sum())

O*NET-SOC Code        0
Title                 0
Achievement           0
Independence          0
Recognition           0
Relationships         0
Support               0
Working Conditions    0
dtype: int64


In [15]:
wv_wide.to_csv("../../notebooks/formatting_onet/formatted_work_values.csv", index=False)

In [16]:
wv_wide.describe()

,Achievement,Independence,Recognition,Relationships,Support,Working Conditions
count,874.000000,874.000000,874.000000,874.00000,874.00000,874.000000
mean,4.119531,4.367494,3.636751,4.45103,4.37587,4.075286
std,1.397540,1.194481,1.305699,1.14019,0.94642,1.131979
min,1.000000,1.670000,1.000000,1.67000,1.67000,1.500000
25%,3.000000,3.330000,2.330000,3.67000,3.67000,3.170000
50%,4.330000,4.670000,3.670000,4.33000,4.33000,4.170000
75%,5.330000,5.330000,4.670000,5.33000,5.00000,5.000000
max,7.000000,7.000000,7.000000,7.00000,7.00000,6.500000


In [17]:
print(wv_wide["O*NET-SOC Code"].duplicated().sum()) 

0


In [18]:
check = pd.read_csv("../../notebooks/formatting_onet/formatted_work_values.csv")
print(check.shape)
check.head()

(874, 8)


,O*NET-SOC Code,Title,Achievement,Independence,Recognition,Relationships,Support,Working Conditions
0,11-1011.00,Chief Executives,6.33,7.00,7.00,5.00,5.33,6.33
1,11-1011.03,Chief Sustainability Officers,6.67,6.67,6.00,5.00,3.33,6.33
2,11-1021.00,General and Operations Managers,5.33,6.00,5.67,6.33,4.67,6.00
3,11-1031.00,Legislators,5.33,5.00,5.00,5.67,4.00,4.33
4,11-2011.00,Advertising and Promotions Managers,5.33,5.33,5.33,5.00,4.00,5.33


O*NET Work Values: cleaning summary

* Raw file: 7,866 rows (Scale ID mixes EX mean ratings and VH high-point flags)

* Filtered to EX only → 5,244 rows

* Dropped Element ID, Scale ID, Scale Name, Date, Domain Source

* Pivoted to wide format → 874 occupations × 6 Work Value elements

* No missing values in the final table; all values fall within the expected 1–7 Extent scale

* Saved as formatted_work_values.csv
